src/peerstate.rs

```diff
@@ -311,6 +362,7 @@ impl Peerstate {
         }
     }
 
+    /// Converts the peerstate into key.
```

Comment:
who's key?  when can it be `None`?  does this modify the peerstate in the database?